llvm/lib/Transforms/Scalar/GVNSink.cpp

```diff
@@ -766,6 +807,9 @@ unsigned GVNSink::sinkBB(BasicBlock *BBEnd) {
              BBEnd->printAsOperand(dbgs()); dbgs() << "\n");
   SmallVector<BasicBlock *, 4> Preds;
   for (auto *B : predecessors(BBEnd)) {
+    // Bailout on malformed CFG where BasicBlock has no predecessor(PR42346).
```

Comment:
Malformed is an exaggeration; the IR verifier is perfectly fine with code like this.

Realistically, it'll almost always get cleaned up before this pass runs, so it doesn't really matter what you do here, but please use clear wording.

Reply:
will do.